### PR TITLE
Add method to API that takes STPPaymentIntentParams and confirms them

### PR DESCRIPTION
**Pull request by danj-stripe**
## Summary
Adding support for `/v1/payment_intents/:id/confirm`

This method takes a STPPaymentIntentParams object, which may have STPSourceParams or a
Source id, among other fields.

## Motivation
IOS-792

## Testing
Added tests. This is part of a series of PRs, starting with #985 (which this relies on),
and this was also tested in the sample app.

### CODE REVIEW

**Comment by danj-stripe** [comment]
This also needs review.

File: Stripe/STPPaymentIntentParams.m [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+//  STPPaymentIntentParams.m
+//  Stripe
+//
+//  Created by Daniel Jackson on 7/3/18.
+//  Copyright © 2018 Stripe, Inc. All rights reserved.
+//
+
+#import "STPPaymentIntentParams.h"
+#import "STPPaymentIntent+Private.h"
+
```
Comment (csabol-stripe) [inline Stripe/STPPaymentIntentParams.m]: I find it useful to include NS_ASSUME_NONNULL_BEGIN/END in the .m files as well. It makes sure your implementation matches your interface!

File: Tests/Tests/STPPaymentIntentParamsTest.m [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@interface STPPaymentIntentParamsTest : XCTestCase
+
+@end
+
+@implementation STPPaymentIntentParamsTest
+
+- (void)testInit {
+    for (STPPaymentIntentParams *params in @[[[STPPaymentIntentParams alloc] initWithClientSecret:@"secret"],
+                                             [[STPPaymentIntentParams alloc] init],
+                                             [STPPaymentIntentParams new],
```
Comment (csabol-stripe) [inline Tests/Tests/STPPaymentIntentParamsTest.m]: no personal preference but mixing `alloc] init]` and `new]` right next to each other reads a bit weird :P

Comment (csabol-stripe) [reply Tests/Tests/STPPaymentIntentParamsTest.m]: Is this explicit to test that `new` actually calls `alloc] init]`? Seems like that's out of scope of something we need to test

File: Stripe/PublicHeaders/STPPaymentIntentParams.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ */
+@property (nonatomic, copy, nullable, readwrite) NSString *receiptEmail;
+
+/**
+ `@YES` to save this PaymentIntent’s Source to the associated Customer,
+ if the Source is not already attached.
+
+ This should be a boolean NSNumber, so that it can be `nil`
+ */
+@property (nonatomic, strong, nullable, readwrite) NSNumber *saveSourceToCustomer;
```
[the remark this answers is not in the document]
Comment (csabol-stripe) [reply Stripe/PublicHeaders/STPPaymentIntentParams.h]: 👍

File: Stripe/STPPaymentIntentParams.m [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+@implementation STPPaymentIntentParams
```
[the remark this answers is not in the document]
Comment (csabol-stripe) [reply Stripe/STPPaymentIntentParams.m]: yeah noticed that later so that makes sense

File: Tests/Tests/STPPaymentIntentParamsTest.m (hunk 2 above)
Comment (csabol-stripe) [reply Tests/Tests/STPPaymentIntentParamsTest.m]: 👍

**Comment by csabol-stripe** [comment]
> I think my answer to most of these is "that's the way the existing code does things", so my preference is to land as-is and write up any improvements as separate JIRAs to be prioritized against our other work, but let me know if you think any of them are high enough priority to include.

Sounds good to me. Land away!

**Pull request merged**